index.js

```diff
@@ -14,16 +30,78 @@ module.exports = postcss.plugin('alter-color', options => {
 	const initialColor = parseColor(options.from);
 	const finalColor = parseColor(options.to);
 
-	console.log(initialColor);
-	console.log(finalColor);
-
-	return (root, result) => {
+	return (root) => {
 		root.walkRules(rule => {
 			rule.walkDecls(decl => {
 				if (canContainColor(decl.prop)) {
 					const parsedValue = cssTree.parse(decl.value, {context: 'value'});
 
-					console.log(JSON.stringify(parsedValue));
+					cssTree.walk(parsedValue, {
+						visit: 'Identifier',
+						enter(node) {
+							if (node.name === initialColor.keyword) {
+								if (finalColor.keyword === undefined) {
+									node.name = finalColor.hex;
+								} else {
+									node.name = finalColor.keyword;
+								}
+							}
+						}
+					});
+
+					cssTree.walk(parsedValue, {
+						visit: 'HexColor',
+						enter(node) {
+							if (node.value.length > 3) {
+								if (node.value === removeHashSymbol(initialColor.hex)) {
+									node.value = removeHashSymbol(finalColor.hex);
+								}
+							} else {
+								if (node.value === convertToShortHex(removeHashSymbol(initialColor.hex))) {
+									node.value = convertToShortHex(removeHashSymbol(finalColor.hex));
+								}
+							}
+						}
+					});
+
+					cssTree.walk(parsedValue, {
+						visit: 'Function',
+						enter(node) {
+
+							const ast = cssTree.toPlainObject(node);
```

Comment:
You don't need this conversion because `node.children` has `.forEach()` and `.map()` methods. Please, see https://github.com/csstree/csstree/blob/master/lib/utils/list.js